### DISCUSSION TITLE
How to retrieve the average rating of a release (or master)?

### DISCUSSION THREAD

**Discussion by amoe** [joalla/discogs_client#101] [Support] [answered]
Hi, great work on the client.

How could I retrieve the average rating?

For instance, this page shows the rating: https://www.discogs.com/master/1837544-Netsky-Second-Nature
Which is 4.6 as of this writing.

I'm also interested in how to fetch the number of users who have a master or release in their collection, does the API make this data available?

**Comment by JOJ0** [comment]
Have a look at the api docs: https://www.discogs.com/developers

If you find the funcionality you are looking for, tell us which one! We might be already supporting it, if not we can make a feature request from this question 😀

**Reply by amoe** [reply]
The info is listed in the example response for a `Release`: https://www.discogs.com/developers#page:database,header:database-release

```
{
    ...
    "community": {
        ...
        "rating": {
            "average": 3.42,
            "count": 45
        },
        ...
    },
    ...
}
```

**Comment by prcutler** [comment]
Looking at the Release object, I don't believe we capture the community info available.  

A dir() shows;

```
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_known_invalid_keys', 'artists', 'changes', 'client', 'companies', 'country', 'credits', 'data', 'data_quality', 'delete', 'fetch', 'formats', 'genres', 'id', 'images', 'labels', 'marketplace_stats', 'master', 'notes', 'price_suggestions', 'refresh', 'save', 'status', 'styles', 'thumb', 'title', 'tracklist', 'url', 'videos', 'year']
```

Looking at release # 1026691 (The Mix-Up by Beastie Boys), the JSON has the following fields in community (I deleted most of the contributors to keep the JSON sample small):
```
    "community": {
        "have": 2636,
        "want": 942,
        "rating": {
            "count": 274,
            "average": 4.38
        },
        "submitter": {
            "username": "Redeamer_NL",
            "resource_url": "https://api.discogs.com/users/Redeamer_NL"
        },
        "contributors": [
            {
                "username": "Redeamer_NL",
                "resource_url": "https://api.discogs.com/users/Redeamer_NL"
            },
        ],
        "data_quality": "Needs Vote",
        "status": "Accepted"
```

I think there is value in adding a feature request to implement the community fields, especially the have / want / rating.

--Paul

**Reply by JOJ0** [reply]
Yeah probably layin them out as attributes would be nice @prcutler 
As a workaround for now: Can you access those things via the .data attribute?

**Comment by prcutler** [comment]
I don't think so (if I'm doing it right).  It looks like .data only returns the Discogs ID & API URL for a given release:

`print(client.release(release).data, dir(client.release(release).data))`

```
{'id': 1026691, 'resource_url': 'https://api.discogs.com/releases/1026691'} ['__call__', '__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__name__', '__ne__', '__new__', '__qualname__', '__reduce__', '__reduce_ex__', '__repr__', '__self__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__text_signature__']
```

**Reply by AnssiAhola** [reply]
If you use the `fetch` method you can access the community data
```python
print(release.fetch("community")["rating"])
>>> {'count': 8, 'average': 4.5}
```

**Reply by amoe** [reply]
This answers my question, thank you :)

**Reply by JOJ0** [reply]
@amoe whoops we haven't gotten around to merge it but you could try out the new feature in this pr: https://github.com/joalla/discogs_client/pull/105

**Comment by JOJ0** [comment, marked as answer]
#105 is merged and released by now.
